lib/ja_serializer/builder/relationship.ex

```diff
@@ -17,10 +17,14 @@ defmodule JaSerializer.Builder.Relationship do
   end
 
   defp add_links(relation, {_type, _name, opts}, context) do
-    case opts[:link] do
-      nil ->  relation
-      path -> Map.put(relation, :links, [Link.build(context, :related, path)])
-    end
+    Keyword.get(opts, :relationship_links, [])
+      |> Keyword.put_new(:related, opts[:link])
+      |> Enum.filter(fn {_key, value} -> is_binary(value) end)
```

Comment:
I don't believe we need this filter. Atoms are also allowed (to use a function to return the link). I also have been meaning to add support for anonymous functions.